python/res/enkf/jobs/enkf_scaling_job.py

```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+
+import sys
+import configsuite
+
+from copy import deepcopy
+from collections import namedtuple
+
+from res.enkf.jobs import enkf_scaling_job_config
+from res.enkf.jobs.enkf_scaling_pca import get_scaling_factor
+from res.enkf import LocalObsdata, ActiveList
+from res.enkf.jobs.enkf_scaling_get_data import get_measured_data, get_scaled_matrix
+
+
+def main_entry_point(ert, user_config_dict):
+    """
+    Takes an instance of EnkFMain and a user config dict, will do some pre-processing on
+    the user config dict, set up a ConfigSuite instance and validate the job before control
+    is passed to the main job.
+    """
+
+    config_dict = _find_and_expand_wildcards(ert, user_config_dict)
+
+    config = setup_configuration(config_dict, enkf_scaling_job_config._build_schema())
+
+    if not valid_configuration(config):
+        sys.exit(1)
+
+    if not valid_job(ert.getObservations(), config):
+        sys.exit(1)
+
+    enkf_scale_job(ert, config)
+
+
+def enkf_scale_job(ert, config):
+    """
+    Collects data, performs scaling and applies scaling, assumes validated input.
+    """
+    obs = ert.getObservations()
+
+    calculate_data = get_measured_data(ert, config.snapshot.CALCULATE_KEYS)
```

Comment:
`calculate_data` should be its own class `MeasuredData` and it should be constructed not with the `ert` object, but `ert.getObservations()` and the calculate keys per princicple of least knowledge.

Reply:
I completely agree, it's just that I'm depending on, among others,  `SummaryObservationCollector` and `SummaryCollector` which both use ert. They should probably be refactored to have less knowledge. There are a couple of issues on an improved data api which should improve this: https://github.com/equinor/libres/issues/675